src/modules/jcms_notifications/src/FetchArticleService.php

```diff
@@ -63,6 +63,43 @@ public function getArticle(string $id) {
   }
 
   /**
+   * Gets the IDs for every article in Lax.
+   *
+   * @return array
+   */
+  public function getAllArticleIDs() {
+    $ids = [];
+    $endpoint = Settings::get('jcms_all_articles_endpoint');
+    if ($endpoint) {
+      $auth_key = 'jcms_article_auth_unpublished';
+      $stop = FALSE;
+      while (!$stop) {
+        $page = $page ?? 1;
+        $response = $this->client->get($endpoint, ['auth' => Settings::get($auth_key), 'http_errors' => FALSE, ['query' => ['page' => $page]]]);
+        if ($response instanceof ResponseInterface) {
+          $body = $response->getBody()->getContents();
+          $json = json_decode($body, TRUE);
+          if (isset($json['items']) && !empty($json['items'])) {
+            foreach ($json['items'] as $data) {
+              if (isset($data['id'])) {
+                $ids[] = $data['id'];
+                $page++;
```

Comment:
@stephenwf It's the pagination variable as we have to go through each page of results. I'll have a look and move it if necessary.